File: 0_1.baekjoon/greedy/java/BOJ16953.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
public class BOJ16953 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());

        int A = Integer.parseInt(st.nextToken());
        int B = Integer.parseInt(st.nextToken());

        System.out.println(getMinimumOperations(A, B));
    }

    private static int getMinimumOperations(int start, int end) {
        int number = end;

        for (int i=0;i<end;i++) {
            if (number == start)
                return i + 1;
                
            if (number % 10 == 1) {
                number /= 10;
            } else if (number % 2 == 0){
                number /= 2;
            } else {
                return -1;
            }
        }

        return -1;
    }
}

File: 0_1.baekjoon/greedy/kotlin/src/BOJ16953.kt
import java.util.*

fun main() = with(System.`in`.bufferedReader()) {
    val st = StringTokenizer(readLine())
    val a: Int = st.nextToken().toInt()
    val b: Int = st.nextToken().toInt()

    print(getMinimumOperations(a, b))
}

private fun getMinimumOperations(a: Int, b: Int): Int {
    var target: Int = b

    for (i in 1..b) {
        if (target == a)
            return i

        target /= if (target % 10 == 1) {
            10
        } else if (target % 2 == 0) {
            2
        } else {
            return -1
        }
    }
    return -1
}